### PR TITLE
add loongarch64 support

### PR DESCRIPTION
**Pull request by qiangxuhui**
The LoongArch architecture (LoongArch) is an Instruction Set Architecture (ISA) that has a RISC style.

Documentations:
ISA:
https://loongson.github.io/LoongArch-Documentation/LoongArch-Vol1-EN.html
ABI:
https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html
More docs can be found at:
https://loongson.github.io/LoongArch-Documentation/README-EN.html

### CODE REVIEW

**Comment by NathanFreeman** [comment]
You need to modify `config.m4` to support `loongarch64` support.

**Comment by qiangxuhui** [comment]
> You need to modify `config.m4` to support `loongarch64` support.

ok, i will do it.

**Comment by qiangxuhui** [comment]
@NathanFreeman Please do review the code changes. Thank you so much!

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ *  | 128 | 132 | 136 | 140 | 144 | 148 | 152 | 156 |  *
+ *  -------------------------------------------------  *
+ *  |    S8     |    FP     |     RA    |     PC    |  *
+ *  -------------------------------------------------  *
+ *                                                     *
+ * *****************************************************/
+
+.file "jump_loongarch64_sysv_elf_gas.S"
+.text
+.globl jump_fcontext
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S:43]: .globl swoole_make_fcontext

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+.align 2
+.type jump_fcontext,@function
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S:45]: .type swoole_make_fcontext,@function

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+jump_fcontext:
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S:46]: swoole_make_fcontext:

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    # load PC
+    ld.d  $a2, $sp, 152
+
+    # restore stack
+    addi.d  $sp, $sp, 160
+
+    # jump to context
+    jr  $a2
+.size jump_fcontext, .-jump_fcontext
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S:118]: .size swoole_make_fcontext,.-swoole_make_fcontext

File: thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    # return pointer to context-data
+    jr  $ra
+
+finish:
+    # exit code is zero
+    li.d  $a0, 0
+    # call _exit(0)
+    b  %plt(_exit)
+
+.size make_fcontext, .-make_fcontext
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S:70]: .size swoole_make_fcontext,.-swoole_make_fcontext

File: thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ *  -------------------------------------------------  *
+ *                                                     *
+ * *****************************************************/
+
+.file "make_loongarch64_sysv_elf_gas.S"
+.text
+.globl make_fcontext
+.align 2
+.type make_fcontext,@function
+make_fcontext:
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S:46]: swoole_make_fcontext:

File: thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+ *  | 128 | 132 | 136 | 140 | 144 | 148 | 152 | 156 |  *
+ *  -------------------------------------------------  *
+ *  |    S8     |    FP     |     RA    |     PC    |  *
+ *  -------------------------------------------------  *
+ *                                                     *
+ * *****************************************************/
+
+.file "make_loongarch64_sysv_elf_gas.S"
+.text
+.globl make_fcontext
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S:43]: .globl swoole_make_fcontext

File: thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+.align 2
+.type make_fcontext,@function
```
Comment (NathanFreeman) [inline thirdparty/boost/asm/make_loongarch64_sysv_elf_gas.S:45]: .type swoole_make_fcontext,@function

**Review by NathanFreeman — COMMENTED** [review]
You need to rename sone function to swoole_make*

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S (hunk 3 above)
Comment (qiangxuhui) [reply thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S]: done.

File: thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S (hunk 2 above)
Comment (qiangxuhui) [reply thirdparty/boost/asm/jump_loongarch64_sysv_elf_gas.S]: done.

**Comment by qiangxuhui** [comment]
done. @NathanFreeman

**Comment by matyhtf** [comment]
@qiangxuhui This is the latest version of boost context assembly code, which is not compatible with the current version used by swoole.

**Comment by qiangxuhui** [comment]
@matyhtf Which version of boost should I use?

> @qiangxuhui This is the latest version of boost context assembly code, which is not compatible with the current version used by swoole.

**Pull request closed without merging**